lib/src/rules/unreachable_from_main.dart

```diff
@@ -592,11 +596,16 @@ extension on Declaration {
 
 extension on AstNode {
   bool get isInExternalVariableTypeOrFunctionReturnType {
-    var ancestorTypeAnnotation = thisOrAncestorOfType<TypeAnnotation>();
-    if (ancestorTypeAnnotation == null) return false;
+    AstNode topTypeAnnotation = this;
+    var parent = this.parent;
+    while (parent is TypeAnnotation) {
+      topTypeAnnotation = parent;
+      parent = topTypeAnnotation.parent;
+    }
```

Comment:
Good idea. I've extracted it.

> Oh, it might be also useful for to check its presence inside a TypeArgumentList above?

Hmm, I'm not sure. The whole "fix" here is a little contrived because I think the issue this is fixing is contrived. The user there purports to use dummy classes for type protection, even when never instantiating them.

Instead of the user's `class Point<Coordinate> {`, I think it is much more idiomatic to have `class Point`, `class GlobalPoint implements Point`, and `class LocalPoint implements Point`.

So... given that I don't think "using a type only as a type argument" is very non-idiomatic, I don't have good examples of what this PR is fixing.

But I took the issue as a broad, "if the type is found _anywhere_ (deeply) in a type argument, it can trigger this and that compile-time errors, so it should be considered 'used'."